core/src/test/scala/org/apache/spark/ui/UISuite.scala

```diff
@@ -424,6 +424,24 @@ class UISuite extends SparkFunSuite {
     }
   }
 
+  test("SPARK-36237: Attach and start handler after application started in UI ") {
+    val conf = new SparkConf()
+      .setMaster("local")
+      .setAppName("test")
+      .set(UI.UI_ENABLED, false)
+    val sc = new SparkContext(conf)
```

Comment:
Is the test failure in GA related this the changes here?

Reply:
> Is the test failure in GA related this the changes here?

Seems so..